tex/generic/pgf/frontendlayer/tikz/libraries/circuits/tikzlibrarycircuits.ee.IEC.code.tex

```diff
@@ -416,10 +416,16 @@
     circuit symbol size=width 2 height 2,
     shape=generic circle IEC,
     /pgf/generic circle IEC/before background={
-      \pgfpathmoveto{\pgfqpoint{0.7pt}{0pt}}
-      \pgfpathquadraticcurveto{\pgfqpoint{ 0.35pt}{-0.5pt}}{\pgfqpoint{0pt}{0pt}}
-      \pgfpathquadraticcurveto{\pgfqpoint{-0.35pt}{ 0.5pt}}{\pgfqpoint{-0.7pt}{0pt}}
-      \pgfusepathqstroke
+      {
+        % First cancel rotation to have straight sine wave
+        \pgfgettransformentries{\a}{\b}{\c}{\d}{\x}{\y}
+        \pgfmathatantwo{\c}{\a}
```

Comment:
Might be better to think about how to disable `transform shape` for the `before background` path, rather than trying to counter an existing rotation. The propose solution might result in chaotic behavior in rotated scopes. Have you tried that?

Reply:
Ah sorry I think I have misunderstood what you proposed. Here I tested removing the `transform shape` from the whole graphic instead of just the `before background` as you said.

If the `transform shape` does only a rotation then disabling it just for the path could work, but I have no idea how to do that... If it can also do for example a scaling, then I guess it should not be disabled (even just for the before background path).

Reply:
Actually I don't think this implementation is appropriate: for some reason I wrote it assuming the transformation matrix is only a combination of rotation, scaling and translation, but it needs not be the case...